casanovo/casanovo.py

```diff
@@ -7,116 +7,294 @@
+    "annotated_peak_path",
+    required=True,
+    nargs=-1,
     type=click.Path(exists=True, dir_okay=False),
 )
-@click.option(
-    "--peak_path",
+def evaluate(
+    annotated_peak_path: Tuple[str],
+    model: Optional[str],
+    config: Optional[str],
+    output: Optional[str],
+    verbosity: str,
+) -> None:
+    """Evaluate de novo peptide sequencing performance.
+
+    ANNOTATED_PEAK_PATH must be one or more annoated MGF files,
+    such as those provided by MassIVE-KB.
+    """
+    output, config = setup(model, config, output, verbosity, False)
+    with ModelRunner(config, model) as runner:
+        logger.info("Sequencing and evaluating peptides from:")
+        for peak_file in annotated_peak_path:
+            logger.info("  %s", peak_file)
+
+        runner.evaluate(annotated_peak_path)
+
+    logger.info("DONE!")
+
+
+@main.command(cls=_SharedParams)
+@click.argument(
+    "train_peak_path",
     required=True,
-    help="The file path with peak files for predicting peptide sequences or "
-    "training Casanovo.",
-    multiple=True,
+    nargs=-1,
+    type=click.Path(exists=True, dir_okay=False),
 )
 @click.option(
-    "--peak_path_val",
-    help="The file path with peak files to be used as validation data during "
-    "training.",
+    "-p",
+    "--validation_peak_path",
+    help="""
+    An annotated MGF file for validation, like from MassIVE-KB. Use this
+    option multiple times to specify multiple files.
+    """,
+    required=True,
     multiple=True,
-)
-@click.option(
-    "--config",
-    help="The file name of the configuration file with custom options. If not "
-    "specified, a default configuration will be used.",
     type=click.Path(exists=True, dir_okay=False),
 )
+def train(
+    train_peak_path: Tuple[str],
+    validation_peak_path: Tuple[str],
+    model: Optional[str],
+    config: Optional[str],
+    output: Optional[str],
+    verbosity: str,
+) -> None:
+    """Train a Casanovo model on your own data.
+
+    TRAIN_PEAK_PATH must be one or more annoated MGF files, such as those
+    provided by MassIVE-KB, from which to train a new Casnovo model.
+    """
+    output, config = setup(model, config, output, verbosity, True)
+    with ModelRunner(config, model) as runner:
+        logger.info("Training a model from:")
+        for peak_file in train_peak_path:
+            logger.info("  %s", peak_file)
+
+        logger.info("Using the following validation files:")
+        for peak_file in validation_peak_path:
+            logger.info("  %s", peak_file)
+
+        runner.train(train_peak_path, validation_peak_path)
+
+    logger.info("DONE!")
+
+
+@main.command()
+def version() -> None:
+    """Get the Casanovo version information"""
+    versions = [
+        f"Casanovo: {__version__}",
+        f"Depthcharge: {depthcharge.__version__}",
+        f"Lightning: {lightning.__version__}",
+        f"PyTorch: {torch.__version__}",
+    ]
+    sys.stderr.write("\n".join(versions) + "\n")
```

Comment:
**question:** Why is this written to stderr instead of using the logger?

Reply:
The main reason is that this command is isolated and occurs outside of setting up logging in the `setup()` function. I guess the other question is should this actually go to stdout, since it is the expected output of the command?

Reply:
Ok, makes sense. Although maybe stdout is indeed more appropriate then?